### PR TITLE
add login with 'remember me' option by storing data in $home/.tiledb/cloud.json

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**
This PR enables TileDB-Cloud-Java to use ```$home/.tiledb/cloud.json``` to store login information. 

To achieve that I created a new Login class. This class is used the first time someone wants to login. This Login class takes as input all credentials and server settings required and has two additional options.
1. ```RememberMe``` : if true, it will store the credentials in ```$home/.tiledb/cloud.json``` for future use. 
2. ```OverwritePrevious```: If true, it will favor the new input credentials over the saved ones. In combination with ```RememberMe``` it can be used to overwrite the data in ```$home/.tiledb/cloud.json```

In Addition I created the TileDBClient class to wrap the ApiClient class, similarly to what we do in python and R to avoid modifying auto-generated classes.

Loading the credentials is done by statically initiallizing the ```TileDBClient``` class.

Note: See updated example file for details.

### CODE REVIEW

File: .openapi-generator-ignore [hunk 1]
```diff
@@ -27,3 +27,7 @@ build.gradle
 src/main/java/org/openapitools/client/model/FilterType.java
 README.md
 settings.gradle
+
+#the api client contains the custom code for static initialization from the json file
+src/main/java/io/tiledb/cloud/rest_api/ApiClient.java
```
Comment (Shelnutt2) [inline .openapi-generator-ignore:32]: I don't think we should override generated classes. Can we take the approach we had in TileDB-Cloud-Py in that we create higher level classes that wrap the generated API?

Comment (DimitrisStaratzis) [reply .openapi-generator-ignore]: Created the TileDBClient class to wrap the autogenerated one

File: src/main/java/io/tiledb/cloud/rest_api/Login.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package io.tiledb.cloud.rest_api;
```
Comment (Shelnutt2) [inline src/main/java/io/tiledb/cloud/rest_api/Login.java:1]: Should we move this up a level so we keep autogenerated in a separate package?

**Review by Shelnutt2 — APPROVED** [review]
Great work! One question then lgtm.

File: src/main/java/io/tiledb/cloud/rest_api/Login.java (hunk 2 above)
Comment (DimitrisStaratzis) [reply src/main/java/io/tiledb/cloud/rest_api/Login.java]: Done!

**Pull request merged**